File: store/src/main/java/com/example/store/Service/stockManagment/interfaces/CharacteristicValueService.java
package com.example.store.Service.stockManagment.interfaces;


import com.example.store.DTO.stockManagment.CharacteristicValueDTO;
import com.example.store.Model.StockMangement.CharacteristicValue;

import java.util.List;
import java.util.Map;
import java.util.Optional;

public interface CharacteristicValueService {

    CharacteristicValue saveCharacteristicValue(CharacteristicValueDTO characteristicValueDTO);

    CharacteristicValue findCharacteristicValueById(Long characteristicValueId);

    List<CharacteristicValue> fetchCharacteristicValueList();

    CharacteristicValue updateCharacteristicValue(CharacteristicValueDTO characteristicValueDTO,Long characteristicValueId);

    void deleteCharacteristicValueById(Long characteristicValueId);

    List<CharacteristicValue>  saveAll(List<CharacteristicValueDTO> characteristicValues);

   // Map<String, String> findCharacteristicValueByProductVariantId(Long productVariantId);

}

File: Frontend/src/app/core/services/stockManagment/CharacteristicValueService/characteristic-value.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { CharacteristicValue } from '../../../../shared/models/StockManagment/CharacteristicValue.model';
import { Observable } from 'rxjs';
import { CharacteristicValueDTO } from '../../../../shared/models/dto/stockManagmentDTO/CharacteristicValue.dto';

@Injectable({
  providedIn: 'root'
})
export class CharacteristicValueService {

  private apiUrl = 'http://localhost:8080/api/characteristicValue';
  private http = inject(HttpClient);

  getAllCharacteristicValue():Observable<CharacteristicValue[]>{
    return this.http.get<CharacteristicValue[]>(`${this.apiUrl}/ListCharacteristicValues`);
  }

  addCharacteristicValue(category : CharacteristicValueDTO):Observable<CharacteristicValue>{
    console.log("Characteristic service is called to add a new Characteristic");
    return this.http.post<CharacteristicValue>(`${this.apiUrl}/addCharacteristicValue`,category);
  }

  findCharacteristicValueById(id:number):Observable<CharacteristicValue>{
    return this.http.get<CharacteristicValue>(`${this.apiUrl}/find/${id}`);
  }

  editCharacteristicValue(characteristicValue:CharacteristicValueDTO , characteristicValueId:number):Observable<CharacteristicValue>{
    return this.http.put<CharacteristicValue>(`${this.apiUrl}/update/${characteristicValueId}`,characteristicValue);
  }

  deleteCharacteristicValue(characteristicValueId:number):Observable<string>{
    return this.http.delete<string>(`${this.apiUrl}/delete/${characteristicValueId}`);
  }

  //save Values 
  saveAllCharacteristicValue(characteristics : CharacteristicValueDTO[]):Observable<CharacteristicValue[]>{
    return this.http.post<CharacteristicValue[]>(`${this.apiUrl}/save-all`,characteristics);
  }
/*
  findCharacteristicValueByProductVariantId(productVariantId:number):Observable<{characteristicName:String,characteristicValue:String}[]>{
    return this.http.get<{characteristicName:String,characteristicValue:String}[]>(`${this.apiUrl}/allCharacteristicValuesByProductVariantId/${productVariantId}`);
  }
  */
}
